From be33e799d968a073c0a8856c96d11e8594b88bc4 Mon Sep 17 00:00:00 2001
From: Yannick Gicquel <ygicquel@gmail.com>
Date: Thu, 22 Nov 2012 07:37:26 +0000
Subject: panel: Fix launcher icon animation ending with black artifact

We simply need to hide the animation window before unref'ing the pixbuf.

https://bugzilla.gnome.org/show_bug.cgi?id=688618
---
diff --git a/gnome-panel/xstuff.c b/gnome-panel/xstuff.c
index 5acbf66..4f8b605 100644
--- a/gnome-panel/xstuff.c
+++ b/gnome-panel/xstuff.c
@@ -326,12 +326,13 @@ zoom_draw (GtkWidget *widget,
 			g_source_remove (zoom->timeout_id);
 		zoom->timeout_id = 0;
 
+		gtk_widget_hide (widget);
+		g_idle_add (idle_destroy, widget);
+
 		g_object_unref (zoom->pixbuf);
 		zoom->pixbuf = NULL;
 
 		g_slice_free (CompositedZoomData, zoom);
-
-		g_idle_add (idle_destroy, widget);
 	} else {
 		GdkPixbuf *scaled;
 		int width, height;
--
cgit v0.9.1
